File: 541-reverse-string-ii/reverse-string-ii.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var reverseStr = function(s, k) {

    s = s.split('')

    for(let i=0;i<s.length;i+=2*k) {
       //1st loop for jump to 2k elements
        let n = k;
        let mid = Math.floor(n/2);

    // 2nd loop for swap processing with starting k elements accessed by x...
        for(let j=0;j<mid;j++){   //just add i,we know this logic by default...

            let temp = s[i+j];
            s[i+j] = s[i+n-(j+1)];
            s[i+n-(j+1)] = temp;

        }
    }
    return s.join('');

     
    
};

File: 541-reverse-string-ii/reverse-string-ii.java
class Solution {
    public String reverseStr(String s, int k) {

        char[] arr = s.toCharArray();

        for(int i=0;i<arr.length;i+=2*k){

            int n = Math.min(k, arr.length - i);
            int mid = n/2;


            for(int j=0;j<mid;j++){

                char temp = arr[i+j];
                arr[i+j] = arr[i+n-(j+1)];
                arr[i+n-(j+1)] = temp;

            }

        }
        return new String(arr);

        
    }
}